templates/openstackbaremetalset/cloudinit/networkdata

```diff
@@ -3,13 +3,16 @@ links:
   id: {{ .CtlplaneInterface }}
   type: vif
 networks:
-- netmask: {{ .CtlplaneNetmask }}
-  link: {{ .CtlplaneInterface }}
+- link: {{ .CtlplaneInterface }}
   id: {{ .CtlplaneInterface }}
+  type: ipv{{ .CtlplaneIpVersion }}
   ip_address: {{ .CtlplaneIp }}
-  type: ipv4
+  netmask: {{ .CtlplaneNetmask }}
 {{- if (index . "CtlplaneGateway") }}
-  gateway: {{ .CtlplaneGateway }}
+  routes:
+  - network: {{ if eq .CtlplaneIpVersion "6" }}::{{ else }}0.0.0.0{{ end }}
```

Comment:
string comparision won't work, `==` should.

{"name":"openstack-edpm-ipam","namespace":"openstack"}, "namespace": "openstack", "name": "openstack-edpm-ipam", "reconcileID": "46e66f25-2dc8-4611-9fcd-bbedf5b5f412", "error": "template: tmp:13:19: executing \"tmp\" at <eq .CtlplaneIpVersion \"6\">: error calling eq: incompatible types for comparison"}

Reply:
hmm, I can't get `==` to work, is it valid in this templateing language?
But I belive the issue is that `CtlplaneIpVersion` is an integer and `"6"` is a string.

I updated it to store the full string `ipv4` / `ipv6` in CtlplaneIpVersion. That worked in https://gotemplate.io/

Reply:
Marking this as resolved, can see in logs that we have `routes` defined correctly.

https://logserver.rdoproject.org/29/129/67d36f8c6697204cbbfc0f2af967ac59312fc2ed/github-check/openstack-baremetal-operator-crc-podified-edpm-baremetal/facf370/ci-framework-data/logs/192.168.122.100/cloud/instances/dd35c05d-bde4-470c-baf6-7a7c9902b06c/network-config.json

```
{
 "config": [
  {
   "mac_address": null,
   "name": "enp1s0",
   "subnets": [
    {
     "address": "192.168.122.100",
     "ipv4": true,
     "netmask": "255.255.255.0",
     "routes": [
      {
       "gateway": "192.168.122.1",
       "netmask": "0.0.0.0",
       "network": "0.0.0.0"
      }
     ],
     "type": "static"
    }
   ],
   "type": "physical"
  },
  {
   "address": [
    "192.168.122.80"
   ],
   "search": [
    "ctlplane.example.com"
   ],
   "type": "nameserver"
  }
 ],
 "version": 1
}
```

Reply:
Right.. `if eq .CtlplaneIpVersion 6` should have worked.